bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/handler/NikoHomeControlActionHandler.java

```diff
@@ -62,46 +63,22 @@ public NikoHomeControlActionHandler(Thing thing) {
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
-        Bridge nhcBridge = getBridge();
-        if (nhcBridge == null) {
-            updateStatus(ThingStatus.UNINITIALIZED, ThingStatusDetail.BRIDGE_UNINITIALIZED,
-                    "Niko Home Control: no bridge initialized when trying to execute action " + actionId);
+        NikoHomeControlCommunication nhcComm = getCommunication();
+        if (nhcComm == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Niko Home Control: bridge communication not initialized when trying to execute action command "
+                            + actionId);
             return;
         }
-        NikoHomeControlBridgeHandler nhcBridgeHandler = (NikoHomeControlBridgeHandler) nhcBridge.getHandler();
-        if (nhcBridgeHandler == null) {
-            updateStatus(ThingStatus.UNINITIALIZED, ThingStatusDetail.BRIDGE_UNINITIALIZED,
-                    "Niko Home Control: no bridge initialized when trying to execute action " + actionId);
-            return;
-        }
-        NikoHomeControlCommunication nhcComm = nhcBridgeHandler.getCommunication();
 
         // This can be expensive, therefore do it in a job.
         scheduler.submit(() -> {
-            if (nhcComm == null || !nhcComm.communicationActive()) {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
-                        "Niko Home Control: bridge communication not initialized when trying to execute action "
-                                + actionId);
-                return;
+            if (!nhcComm.communicationActive()) {
+                restartCommunication(nhcComm);
             }
 
             if (nhcComm.communicationActive()) {
```

Comment:
No, because communication state may be updated by another thread in between the 2 statements. The purpose here is to try to restart once if the communication was lost. If that fails, the thing will be offline and the command will not be handled.